In [2]:
import pandas as pd

Preperation of the input data:
- Reading in all JSON input files
- adding 1000000*gameNumber to the index to include the game-ID in the tickCOunt
- make tickCount the index
- combine all dataframes


In [5]:
def prepareInputDf():
  input_df = pd.DataFrame(columns=['buttons', 'mouseDeltaX', 'mouseDeltaY'])
  path = 'drive/MyDrive/BA_JSON/'
  files = ['cs02_input.json', 'cs03_input.json', 'cs04_input.json', 
           'cs06_input.json', 'cs07_input.json', 'cs08_input.json', 
           'cs09_input.json', 'cs10_input.json', 'cs12_input.json']
           # 'cs01_input.json', 'cs11_input.json', 'cs13_input.json'
           # These three are missing due to wrong mouse input data
  game_index_factor = 1000000
  game_index = 0
  
  for x in files:
    file = path + x
    df = pd.read_json(file)
    df['tickCount'] += game_index_factor*game_index
    game_index += 1
    df = df.set_index('tickCount')
    df = df[~df.index.duplicated(keep='first')]
    input_df = pd.concat([input_df, df])
  return input_df

In [6]:
input_df = prepareInputDf()

In [7]:
len(input_df)

644957

The dataset has the data of 644957 frames.

Take a look at the created dataset

In [8]:
input_df.head(5)

buttons mouseDeltaX mouseDeltaY
718      []           0           0
719      []           0           0
720      []           0           0
721      []           0           0
722      []           0           0

In [9]:
input_df[500000:500010]

buttons mouseDeltaX mouseDeltaY
6099189          [duck, moveleft]           0           0
6099190  [attack, duck, moveleft]           0           0
6099191  [attack, duck, moveleft]           5           0
6099192  [attack, duck, moveleft]          16          -1
6099193  [attack, duck, moveleft]           3          -4
6099194  [attack, duck, moveleft]          -1         -10
6099195  [attack, duck, moveleft]          -4          -5
6099196        [attack, moveleft]          -6           1
6099197                [moveleft]         -11          11
6099198                [moveleft]          -7          18

Save the dataset as csv

In [10]:
input_df.to_csv('drive/MyDrive/BA_JSON/input_data.csv')